xscen/ensemble.py

```diff
@@ -0,0 +1,86 @@
+import logging
+from pathlib import Path
+from typing import Optional, Union
+
+import pandas as pd
+import xarray as xr
+from xclim import ensembles
+
+from .catalog import ProjectCatalog, generate_id
+
+logger = logging.getLogger(__name__)
+
+
+def ensemble_stats(
+    datasets: list,
+    create_kwargs: dict = {},
+    statistics: str = "ensemble_percentiles",
+    stats_kwargs: dict = {},
+    common_attrs_only: bool = True,
+    to_level: str = "ensemble",
+):
+    """
+    Create ensemble and calculate statistics on it.
+
+    Parameters
+    ----------
+    datasets: list
+        List of file paths or xarray Dataset/DataArray objects to include in the ensemble
+        Tip: With a project catalog, you can do: `datasets = list(pcat.search(**search_dict).df.path)`.
+    create_kwargs: dict
+        Dictionary of arguments for xclim.ensembles.create_ensemble
+    statistics: str
+        Name of the xclim.ensemble function to call on the ensemble
+        (eg. "ensemble_percentiles","ensemble_mean_std_max_min").
+    stats_kwargs: dict
+        Dictionary of arguments for the statictics function
+    common_attrs_only:
+        If True, keeps only the global attributes that are the same for all datasets and generate new id.
+        If False, keeps global attrs of the first dataset (same behaviour as xclim.ensembles.create_ensemble)
+    to_level: str
+        The processing level to assign to the output.
+
+    Returns
+    -------
+    ens_stats: xr.Dataset
+        Dataset with ensemble statistics
+    """
+    logger.info(f"Creating ensemble with {len(datasets)} and calculating {statistics}.")
+
+    # if input files are .zarr, change the engine automatically
+    if isinstance(datasets[0], (str, Path)):
```

Comment:
Should we have a check that all entries are the same type?

Reply:
est-ce que ça va vraiment arrivé de passé un liste mélangée de paths et datasets?

Reply:
Ouin ok, je vois que `xclim.ensembles.create_ensemble` ne fait pas de checks non plus. J'imagine que ça va juste planter si quelqu'un essaie d'envoyer un mélange des deux.